### ISSUE TITLE
Autocomplete: exception thrown when typing in autocomplete before data available

### ISSUE THREAD

**Issue by georgesimms** [ocadotechnology/hexagonjs#176]
Consider:

```
hx.select('.hx-content').append(hx.autoComplete(() => {}))
```
(This is simulating AJAX being very slow.) If you start typing in this box, an exception is thrown : 
```
_.menu.dropdown._.dropdownContent is not a function.
```

**Comment by georgesimms** [comment]
I am not sure how you would test this

**Comment by c-frater** [comment]
I've already fixed this as mentioned yesterday #172

**Comment by c-frater** [comment]
Also, adding tests for this will be done as part of #125
